hudi-utilities/src/main/java/org/apache/hudi/utilities/transform/ChainedTransformer.java

```diff
@@ -112,6 +118,30 @@ private void validateIdentifier(String id, Set<String> identifiers, String confi
     }
   }
 
+  private StructType getExpectedTransformedSchema(TransformerInfo transformerInfo, JavaSparkContext jsc, SparkSession sparkSession,
+                                                  Option<StructType> incomingStructOpt, Option<Dataset<Row>> rowDatasetOpt, TypedProperties properties) {
+    if (!sourceSchemaOpt.isPresent() && !rowDatasetOpt.isPresent()) {
```

Comment:
Should it be an `||` instead of `&&`? Don't we want both schema and row dataset to be present?

Reply:
We need only one of these. If source schema is present we can use it directly else we can infer schema from the row dataset available.

Reply:
Thanks for the clarification.